### PR TITLE
Init audio fix

### PR DESCRIPTION
**Pull request by ThomasWaldmann**

### CODE REVIEW

File: multigeiger/speaker.cpp [hunk 1]
```diff
@@ -167,7 +167,6 @@ void play(int *sequence) {
 
 void setup_speaker(bool playSound, bool _led_tick, bool _speaker_tick) {
   pinMode(LED_BUILTIN, OUTPUT);
-  digitalWrite(LED_BUILTIN, LOW);  // LED off
```
Comment (ThomasWaldmann) [inline multigeiger/speaker.cpp:170]: this is done by playing the TONE(0,0,0,0) now.

File: multigeiger/speaker.cpp [hunk 2]
```diff
@@ -183,27 +182,33 @@ void setup_speaker(bool playSound, bool _led_tick, bool _speaker_tick) {
...
-      TONE(0, 0, -1, 0),        // speaker off, end
-    };
-    play((int *)sequence);
-  }
+  tick_enable(false);  // no ticking while we play melody / init sound
+
+  static int init[][4] = {
+    TONE(0, 0, 0, 0),        // speaker off, led off, end
+  };
+  play((int *)init);
```
Comment (ThomasWaldmann) [inline multigeiger/speaker.cpp:190]: fix for audio issue #398

File: multigeiger/speaker.cpp [hunk 3]
```diff
@@ -183,27 +182,33 @@ void setup_speaker(bool playSound, bool _led_tick, bool _speaker_tick) {
...
+    TONE(1174659, 1, -1, 4),  // D
+    TONE(987767, 1, -1, 2),   // H
+    TONE(1046502, 1, -1, 2),  // C
+    TONE(987767, 1, -1, 4),   // H
+    TONE(987767, 0, -1, 4),   // H
+    TONE(0, 0, -1, 2),        // ---
+    TONE(0, 0, -1, 0),        // speaker off, end
+  };
+  if (playSound)
+    play((int *)melody);
```
Comment (ThomasWaldmann) [inline multigeiger/speaker.cpp:211]: I just changed the 3rd element of all TONE tuples to -1 "don't touch" - resetting the LED to 0 in the tone pauses is neither wanted nor needed.

Diff looks like more than it really is.

File: multigeiger/speaker.cpp [hunk 4]
```diff
@@ -183,27 +182,33 @@ void setup_speaker(bool playSound, bool _led_tick, bool _speaker_tick) {
...
-      TONE(987767, 0, -1, 4),   // H
-      TONE(0, 0, 0, 2),         // ---
-      TONE(0, 0, -1, 0),        // speaker off, end
-    };
-    play((int *)sequence);
-  }
+  tick_enable(false);  // no ticking while we play melody / init sound
+
+  static int init[][4] = {
+    TONE(0, 0, 0, 0),        // speaker off, led off, end
```
Comment (M-a-r-i-a-n-n-e) [inline multigeiger/speaker.cpp:188]: @ThomasWaldmann should we use -1 here, too?

**Review by M-a-r-i-a-n-n-e — APPROVED** [review]

File: multigeiger/speaker.cpp (hunk 4 above)
Comment (ThomasWaldmann) [reply multigeiger/speaker.cpp:188]: no, we really want to init the LED here. see removed line 170.

Comment (M-a-r-i-a-n-n-e) [reply multigeiger/speaker.cpp:188]: I see 👍

Comment (ThomasWaldmann) [reply multigeiger/speaker.cpp:188]: hmm? this is not line 202 in this PR's code.

Comment (M-a-r-i-a-n-n-e) [reply multigeiger/speaker.cpp:188]: No but the // --- lines do not have a -1 there. Is it not needed when the other parameters are 0?

Comment (M-a-r-i-a-n-n-e) [reply multigeiger/speaker.cpp:188]: Ah sorry.. looked into the wrong file

**Pull request merged**